routing/dht/routing.go

```diff
@@ -84,7 +84,10 @@ func (dht *IpfsDHT) GetValue(ctx context.Context, key key.Key) ([]byte, error) {
 	ctx, cancel := context.WithTimeout(ctx, time.Minute)
 	defer cancel()
 
-	vals, err := dht.GetValues(ctx, key, 16)
+	// retrieve a majority of the expected record count
+	majority := (KValue / 2) + 1
```

Comment:
The probability that a user will find a fresh record depends on things like: 
- the size of the network
- the churn of the network
- the freshness of the routing table
- the number of records sought.

and the number of records is (or is roughly like) an exponent on the probability of NOT finding the freshest record. the take away is that a _very small_ number (<10) is not going to give good results in the presence of (even modest) churn, or a not-perfectly-formed routing table. rising above `10` significantly improves the probability.

--- 

please do not change numbers if you do not fully understand the algorithmic implications.

there are other ways to speed up the resolution rather than "not looking". instead we should be:

- cutting down the time for dials
- _ensuring_ that the resolution is only one way so it terminates quickly, if there wont be 16 records. (the more i look at the dht the more i'm convinced it's exhausting the network, not just the subset of the network in the gradient (towards the lowest XOR distance) of the key).

i suspect the slowness here is that the node finds all the records available on the network but still searches through peers that are _not_ in the gradient.